src/main/java/com/exasol/adapter/document/mapping/PropertyToBoolColumnMapping.java

```diff
@@ -36,6 +33,95 @@ public ColumnMapping withNewExasolName(final String newExasolName) {
+        @Override
+        protected B fillValuesFrom(final C instance) {
+            super.fillValuesFrom(instance);
+            PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilder.fillValuesFromInstanceIntoBuilder(instance,
+                    this);
+            return self();
+        }
+
+        private static void fillValuesFromInstanceIntoBuilder(final PropertyToBoolColumnMapping instance,
+                final PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilder<?, ?> b) {
+            b.notBooleanBehavior(instance.notBooleanBehavior);
+        }
+
+        @Override
+        protected abstract B self();
+
+        @Override
+        public abstract C build();
+
+        /**
+         * @param notBooleanBehavior behaviour in case value is not a boolean
+         * @return {@code this}.
+         */
+        public B notBooleanBehavior(final ConvertableMappingErrorBehaviour notBooleanBehavior) {
+            this.notBooleanBehavior = notBooleanBehavior;
+            return self();
+        }
+
+        @Override
+        public String toString() {
+            return "PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilder(super=" + super.toString()
+                    + ", notBooleanBehavior=" + this.notBooleanBehavior + ")";
+        }
+    }
+
+    private static final class PropertyToBoolColumnMappingBuilderImpl extends
+            PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilder<PropertyToBoolColumnMapping, PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilderImpl> {
+
+        private PropertyToBoolColumnMappingBuilderImpl() {
+        }
+
+        @Override
+        protected PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilderImpl self() {
+            return this;
+        }
+
+        /**
+         * Build a new instance.
+         * 
+         * @return new instance
+         */
+        @Override
+        public PropertyToBoolColumnMapping build() {
+            return new PropertyToBoolColumnMapping(this);
+        }
+    }
+
+    /**
+     * Creates a new instance from a builder.
+     * 
+     * @param builder builder
+     */
+    protected PropertyToBoolColumnMapping(
+            final PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilder<?, ?> builder) {
+        super(builder);
+        this.notBooleanBehavior = builder.notBooleanBehavior;
+    }
+
+    /**
+     * Create a new builder for {@link PropertyToBoolColumnMapping}.
+     * 
+     * @return a new builder
+     */
+    @SuppressWarnings("java:S1452") // Generic wildcard type is ok here
+    public static PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilder<?, ?> builder() {
+        return new PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilderImpl();
+    }
+
+    private PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilder<?, ?> toBuilder() {
+        return new PropertyToBoolColumnMapping.PropertyToBoolColumnMappingBuilderImpl().fillValuesFrom(this);
```

Comment:
```suggestion
        return builder().fillValuesFrom(this);
```

Reply:
Not possible, because `fileValuesFrom()` is only available in the concrete impl class